### PR TITLE
Yamale schema validation for data set descriptions

### PR DESCRIPTION
**Pull request by lucasmbrown-usds**
Ready for review. This is part of this ticket: https://app.zenhub.com/workspaces/justice40-60993f6e05473d0010ec44e3/issues/usds/justice40-tool/40. 

This work in Python got a little ahead of some other work we want to do to setup both our Python tooling and our Python best practices (e.g., code style and conventions). That work is [tracked here](https://app.zenhub.com/workspaces/justice40-60993f6e05473d0010ec44e3/issues/usds/justice40-tool/57).

I'm going to propose we split off implementing the first draft of the "actual" data set description schema into a separate PR, to separate concerns and make the reviews smaller.

### CODE REVIEW

**Review by NatHillardUSDS — COMMENTED** [review]
Cool! 
A few thoughts:
* Might overall be helpful for context to link to a relevant user story and describe the goals for the tool
* Curious about the tradeoffs of this approach vs. an existing one like [this](https://github.com/nrkno/yaml-schema-validator-github-action) OOB GHA action? -- could this fit our needs? 
* The template generation is cool - could this be a github [issue template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) as well, and/or write to the `.github/ISSUE_TEMPLATE` folder?

File: data-roadmap/setup.py
Comment (lucasmbrown-usds) [reply data-roadmap/setup.py:20]: Awesome, I love it! Maybe to separate concerns (and be honest about how much time I have available for directly contributing code), maybe we can create a separate ticket to ADR poetry and convert to using it?

**Comment by lucasmbrown-usds** [comment]
> Cool!
> A few thoughts:
> 
> * Might overall be helpful for context to link to a relevant user story and describe the goals for the tool
> * Curious about the tradeoffs of this approach vs. an existing one like [this](https://github.com/nrkno/yaml-schema-validator-github-action) OOB GHA action? -- could this fit our needs?
> * The template generation is cool - could this be a github [issue template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) as well, and/or write to the `.github/ISSUE_TEMPLATE` folder?

Good points! Yes, would love to add a user story. Where's the best place for that @NatHillardUSDS -- in the code, or I've been thinking about adding a readme.md to the python package? 

For the OOB GHA, it wouldn't cover custom logic such as, "make sure every field has a description in the field_description.yaml". But otherwise it's the same. It doesn't seem particularly challenging to run a Python function using a GHA, so I just went with that. 

Yes, creating an issue template would be great!

File: data-roadmap/setup.py
Comment (NatHillardUSDS) [reply data-roadmap/setup.py:20]: Yep this makes sense!

Comment (NatHillardUSDS) [reply data-roadmap/setup.py:20]: (I could also help out with this bit - I've moved almost everything I do in Python-land over to poetry since seeing its benefits)

**Comment by NatHillardUSDS** [comment]
> Good points! Yes, would love to add a user story. Where's the best place for that @NatHillardUSDS -- in the code, or I've been thinking about adding a readme.md to the python package?

Ah, I was thinking on the project board, as part of the "Data Roadmap" epic - I realize this corresponds to [this](https://app.zenhub.com/workspaces/justice40-60993f6e05473d0010ec44e3/issues/usds/justice40-tool/40) card. That said,  the tool itself would also benefit from a general README.md 

> For the OOB GHA, it wouldn't cover custom logic such as, "make sure every field has a description in the field_description.yaml". But otherwise it's the same. It doesn't seem particularly challenging to run a Python function using a GHA, so I just went with that.

This makes sense 👍 

> Yes, creating an issue template would be great!
I guess this is also part of another story

File: data-roadmap/data_roadmap/utils/utils_data_set_description_schema.py
Comment (NatHillardUSDS) [inline data-roadmap/data_roadmap/utils/utils_data_set_description_schema.py:23]: small nit but you can do 
```py
UTILS_DIRECTORY  = DATA_ROADMAP_DIRECTORY / "utils"
```
for this and all the other `joinpath`s -- with `pathlib` its equivalent!

File: data-roadmap/data_roadmap/utils/utils_data_set_description_schema_test.py
Comment (NatHillardUSDS) [inline data-roadmap/data_roadmap/utils/utils_data_set_description_schema_test.py:1]: Great set of tests!

File: data-roadmap/setup.py
Comment (NatHillardUSDS) [reply data-roadmap/setup.py:20]: For future reference this is now captured by [this](https://github.com/usds/justice40-tool/issues/57) card

**Review by NatHillardUSDS — COMMENTED** [review]
Nice! Looks good overall, the last thing I would say is it would help to have a `README` for this / short documentation to set some context on what it is and how to run it - this will help for future reference - with the readme lgtm

File: data-roadmap/data_roadmap/utils/utils_data_set_description_schema.py
Comment (lucasmbrown-usds) [reply data-roadmap/data_roadmap/utils/utils_data_set_description_schema.py:23]: Ooooooh nice! Sexy.

**Review by NatHillardUSDS — DISMISSED** [review]
Awesome, looks good to me!

**Review by NatHillardUSDS — APPROVED** [review]
Ah - Just realized you had pushed the "rearranging imports" commit after my initial approval -- I had turned on "Dismiss stale pull request approvals when new commits are pushed" so it dismissed after this; approving again! As a side note, I typically turn on "Format on save" to save me from having to run it separately -- you may have also done this actually but figured I should mention it! (https://dev.to/adamlombard/how-to-use-the-black-python-code-formatter-in-vscode-3lo0)

**Pull request merged**
